frontend/amundsen_application/static/js/pages/TableDetailPage/index.tsx

```diff
@@ -341,6 +343,50 @@ export class TableDetail extends React.Component<
     );
   }
 
+  renderTableWriterDropdowns(tableWriter, tableApps) {
+    const apps: TableWriter[] = [];
+    if (
+      (tableApps === null || tableApps.length === 0) &&
+      tableWriter === null
+    ) {
+      return null;
+    }
+    if (tableApps !== null && tableApps.length > 0) {
+      apps.push(...tableApps);
+    }
+    if (
+      tableWriter !== null &&
+      !apps.some((app) => app.id === tableWriter.id)
+    ) {
+      apps.push(tableWriter);
+    }
+
+    const writerDropdowns: JSX.Element[] = [];
+    const airflowApps = apps.filter(
+      (app) => app.name.toLowerCase() === AIRFLOW
+    );
+    const databricksApps = apps.filter(
+      (app) => app.name.toLowerCase() === DATABRICKS
+    );
+    const remainingApps = apps.filter(
+      (app) =>
+        app.name.toLowerCase() !== AIRFLOW &&
+        app.name.toLowerCase() !== DATABRICKS
+    );
+
+    if (airflowApps.length > 0) {
+      writerDropdowns.push(<WriterDropdown tableApps={airflowApps} />);
+    }
+    if (databricksApps.length > 0) {
+      writerDropdowns.push(<WriterDropdown tableApps={databricksApps} />);
+    }
+    if (remainingApps.length > 0) {
+      writerDropdowns.push(<WriterDropdown tableApps={remainingApps} />);
+    }
```

Comment:
I think I see what you are trying to do here, but it feels non-idiomatic in the React world.
Check out this resource: https://reactjs.org/docs/conditional-rendering.html

We tend to favor the 'Inline If with Logical && Operator' method nowadays.